SQLQuery(2)_2609.sql

```diff
@@ -0,0 +1,56 @@
+
+CREATE TABLE Sales (
+    SaleID INT PRIMARY KEY,
+    ProductID INT,
+    CustomerID INT,
+    SaleDate DATE,
+    Quantity INT,
+    UnitPrice DECIMAL(10, 2)
+);
+
+CREATE TABLE Products (
+    ProductID INT PRIMARY KEY,
+    ProductName VARCHAR(100),
+    Category VARCHAR(50)
+);
+
+CREATE TABLE Customers (
+    CustomerID INT PRIMARY KEY,
+    CustomerName VARCHAR(100),
+    City VARCHAR(50),
+    State VARCHAR(2)
+);
+*/
+
+Select * from Sales;
+Select * from Products;
+Select * from Customers;
+--Calculate the total sales amount for each product. Display the ProductID and total sales amount.
+SELECT ProductID, (Quantity * UnitPrice) AS SaleTotal FROM Sales;
+
+--Find the average quantity sold per sale for each product category.
+
+SELECT p.Category, AVG(s.Quantity) from Sales s JOIN Products p ON s.ProductID = p.ProductID GROUP BY P.Category;

```

Comment:
What if I have products that are not sold, What will be the output of this query?

Reply:
Since we use inner join between the sales and products table, any product that wasn't sold would not be present in the output as it isnt present in the Sales table (i.e.  inner join only returns records where the joining condition is satisfied in both tables)

Reply:
please alter the query to get all the products with quantity as 0 (if no products sold.)

Reply:
i Have updated the query to return categories which have products that have 0 sales

Reply:
What is the column value for the products which are not sold ? I need 0 not null.

Reply:
Noted and changed the query to include:
 COALESCE (AVG(s.Quantity),0)
This will replace the NULL values with 0 in the output

Reply:
Good. check the difference between ISNNULL vs COALESCE